src/day14/MaxStack.java

```diff
@@ -0,0 +1,38 @@
+package day14;
+
+import java.util.Stack;
+
```

Comment:
Neat, clean, no overrides, very nice!  If you are curious about other approaches, GB and Tara did variations on a different approach that I think you'd find pretty cool.